databroker/assets/handlers.py

```diff
@@ -203,6 +203,105 @@ def __init__(self, filename, frame_per_point=1):
             frame_per_point=frame_per_point)
 
 
+class AreaDetector_HDF5SingleHandler(HandlerBase):
+    '''Handler for hdf5 data stored 1 image per file and returned as a
+    numpy.array.
+
+    Parameters
+    ----------
+    fpath : string
+        filepath
+    template : string
+        filename template string.
+    filename : string
+        filename
+    key : string
+        the 'path' inside the file to the data set.
+    column_names : list[str]
+        The column names of the table
+    frame_per_point : float
+        the number of frames per point.
+    '''
+    specs = {'AD_HDF5_SINGLE'} | HandlerBase.specs
+
+    def __init__(self, fpath, template, filename, key='/entry/data/data',
+                 frame_per_point=1):
+        # I have included defaults for `key` for back compatibility.
+        self._path = os.path.join(fpath, '')
+        self._fpp = frame_per_point
+        self._template = template
+        self._filename = filename
+        self._key = key
+
+    def _fnames_for_point(self, point_number):
+        start = int(point_number * self._fpp)
+        stop = int((point_number + 1) * self._fpp)
+        for j in range(start, stop):
+            yield self._template % (self._path, self._filename, j)
+
+    def __call__(self, point_number):
+        ret = []
+        import h5py
```

Comment:
Maybe import is once in the beginning of the file, so that it won't be necessary to do it twice.

Reply:
I am following the style from all the previous handlers which is to import the filehandler module just before it is first used, for example see [here](https://github.com/NSLS-II/databroker/blob/f4fbfd352d6ab6ffbd494e9e8d948764423abee4/databroker/assets/handlers.py#L171).

If we want to change this I am happy to do it.

Reply:
I see, thanks for the reference. This importing within a function makes it harder to track all external dependencies, so I'd vote for moving it to the top of the file.

Reply:
I think this was our half-baked attempt to isolate dependencies so that, as `handlers.py` grows, we don't accumulate a ton of required dependency in `databroker`. The "proper" fix, in my view, is to follow the suitcase/intake pattern and create many small packages, as we have started to do with `eiger-io`.

I vote to follow @mrakitin here.

Reply:
I like the idea of using separate packages for specific handlers in the future.